web/src/lib/components/asset-viewer/photo-viewer.svelte

```diff
@@ -60,8 +65,20 @@
       });
 
       assetData = URL.createObjectURL(data);
+      imageLoaded = true;
+
+      if (preloadAssets) {
```

Comment:
Could you return if `!preloadAssets` and otherwise continue to avoid such a huge `if`?